File: 이진영/Stack/다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
    let passing = [];
    let t = 0;
    let cur_weight = 0;
    
    while (truck_weights.length > 0 || passing.length > 0) {
        if (passing.length > 0 && t - passing[0][1] >= bridge_length) {
            let wt = passing.shift();
            cur_weight -= wt[0];
        }
        
        if (truck_weights[0] + cur_weight <= weight && passing.length < bridge_length) {
            let w = truck_weights.shift()
            passing.push([w, t]);
            cur_weight += w
        }
        
        t++;
    }
    
    return t;
}

File: 정민선/프로그래머스/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    answer = 0
    bridge = [0] * bridge_length
    check = 0
    
    while truck_weights or check > 0:
        answer += 1
        check -= bridge.pop(0)
        if truck_weights:
            if truck_weights[0] + check <= weight:
                truck = truck_weights.pop(0)
                bridge.append(truck)
                check += truck
            else:
                bridge.append(0)
            
    return answer